### PR TITLE
5 patch/api/reviews/reviewid

### PR DESCRIPTION
**Pull request by mgrindrod92**
Updated code for task 5 - update review votes. 6 new tests added, as well as updated controller and model.

### CODE REVIEW

File: controller/category_controller.js [hunk 1]
```diff
@@ -1,8 +1,17 @@
 const { selectCategories } = require('../model/category_model');
 
 exports.getCategories = (req, res, next) => {
+
     selectCategories()
     .then((categories) => {
-        res.status(200).send( {categories} )
+        if (categories.length === 0) {
+            console.log('Hi');
```
Comment (daviidbartlett) [inline controller/category_controller.js:8]: remove console.log

Comment (mgrindrod92) [reply controller/category_controller.js:8]: Updated now. Thanks

File: controller/category_controller.js [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+            return Promise.reject({
+                status: 404,
```
Comment (daviidbartlett) [inline controller/category_controller.js:10]: I don't know why you've added this error handling. The endpoint is asking _give me all the categories you have_ and so no categories is valid (not an error). This is different to _give me this category_, if it doesn't exist we'd give them an error.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -81,4 +83,60 @@ describe('GET api/reviews/:review_id', () => {
         .then(({ body }) => 
         expect(body.msg).toBe('Not found. This review does not exist'))
     })
+})
+
+describe.only('PATCH api/reviews/:review_id', () => {
```
Comment (daviidbartlett) [inline __tests__/app.test.js:88]: remove .only

File: __tests__/app.test.js [hunk 4]
```diff
@@ -81,4 +83,60 @@ describe('GET api/reviews/:review_id', () => {
...
+    test("200: returns a 200 status with the initial review", () => {
+        return request(app)
+        .patch('/api/reviews/9')
+        .send({ inc_votes: 10 })
+        .expect(200)
+        .then(({ body }) => {
+            expect(typeof body).toBe('object');
```
Comment (daviidbartlett) [inline __tests__/app.test.js:95]: can we be more specific about the shape of this object?

File: __tests__/app.test.js (hunk 3 above)
Comment (mgrindrod92) [reply __tests__/app.test.js:88]: Removed. Thanks

File: __tests__/app.test.js [hunk 5]
```diff
@@ -81,4 +83,60 @@ describe('GET api/reviews/:review_id', () => {
...
+            expect(typeof body).toBe('object');   
+        })
+    })
+    test('200: returns an updated review (with the new number of votes)', () => {
+        return request(app)
+        .patch(`/api/reviews/9`)
+        .send({ inc_votes: 10 })
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.review[0].votes).toBe(20);
```
Comment (daviidbartlett) [inline __tests__/app.test.js:104]: probably wouldn't expect one review to come back in array

File: controller/category_controller.js (hunk 2 above)
Comment (mgrindrod92) [reply controller/category_controller.js:10]: This is something I added when discussing code with Duncan the other day. I can remove this if it's unnecessary

File: __tests__/app.test.js (hunk 4 above)
Comment (mgrindrod92) [reply __tests__/app.test.js:95]: What do you mean? I don't understand the question.

**Review by daviidbartlett — CHANGES_REQUESTED** [review]
Only a few things to change!

**Pull request merged**